spring-vault-core/src/main/java/org/springframework/vault/config/EnvironmentVaultConfiguration.java

```diff
@@ -326,6 +336,67 @@ protected ClientAuthentication awsEc2Authentication() {
 				restOperations());
 	}
 
+	protected ClientAuthentication awsIamAuthentication() {
+
+		AWSCredentialsProvider credentialsProvider = AwsCredentialProvider
+				.getAwsCredentialsProvider();
+
+		AwsIamAuthenticationOptionsBuilder builder = AwsIamAuthenticationOptions
+				.builder();
+
+		String roleId = getProperty("vault.aws-iam.role-id");
+		if (StringUtils.hasText(roleId )) {
+			builder.role(roleId);
+		}
+
+		String serverId = getProperty("vault.aws-iam.server-id");
+		if (StringUtils.hasText(serverId)) {
+			builder.serverName(serverId);
+		}
+
+		String awsPath = getProperty("vault.aws-iam.aws-path");
+		builder.path(awsPath)
+				.credentialsProvider(credentialsProvider);
+
+		AwsIamAuthenticationOptions options = builder
+				.credentialsProvider(credentialsProvider).build();
+
+		return new AwsIamAuthentication(options, restOperations());
+	}
+
+	private static class AwsCredentialProvider {
+
+		private static AWSCredentialsProvider getAwsCredentialsProvider() {
+
+			DefaultAWSCredentialsProviderChain backingCredentialsProvider = DefaultAWSCredentialsProviderChain
+					.getInstance();
+
+			// Eagerly fetch credentials preventing lag during the first, actual login.
+			AWSCredentials firstAccess = backingCredentialsProvider.getCredentials();
+
+			AtomicReference<AWSCredentials> once = new AtomicReference<>(firstAccess);
```

Comment:
Care to elaborate the need for intermediate caching? Typically, `AWSCredentialsProvider` perform caching on their own unless calling `.refresh()`. We should either leave these defaults or use an entirely-static `AWSCredentials` instance.